test/distribution/test_nan_mixture.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import Iterable, List, Tuple
+
+# Third-party imports
+import mxnet as mx
+import numpy as np
+import pytest
+
+# First-party imports
+from gluonts.gluonts_tqdm import tqdm
+from gluonts.model.common import Tensor, NPArrayLike
+from gluonts.mx.distribution.distribution import Distribution
+
+from gluonts.mx.distribution import (
+    Categorical,
+    CategoricalOutput,
+    Gaussian,
+    GaussianOutput,
+    NanMixture,
+    NanMixtureOutput,
+    StudentTOutput,
+)
+from gluonts.core.serde import dump_json, load_json
+
+serialize_fn_list = [lambda x: x, lambda x: load_json(dump_json(x))]
+
+
+def diff(x: NPArrayLike, y: NPArrayLike) -> np.ndarray:
+    return np.mean(np.abs(x - y))
+
+
+NUM_SAMPLES = 2_000
+NUM_SAMPLES_LARGE = 100_000
+
+p = np.array([[[0.0001, 0.0001, 0.5]], [[0.999, 0.999, 0.5]]])
+p_cat = np.array(
+    [
+        [[[0.1, 0.9], [0.9, 0.1], [0.5, 0.5]]],
+        [[[0.9, 0.1], [0.01, 0.99], [0.45, 0.55]]],
+    ]
+)
+SHAPE = p.shape
+
+
+@pytest.mark.parametrize(
+    "distr, cat_distr, p, p_cat",
+    [
+        (
+            Gaussian(
+                mu=mx.nd.zeros(shape=SHAPE),
+                sigma=1e-3 + mx.nd.ones(shape=SHAPE),
+            ),
+            Categorical(log_probs=mx.nd.log(mx.nd.array(p_cat))),
+            mx.nd.array(p),
+            mx.nd.array(p_cat),
+        ),
+    ],
+)
+@pytest.mark.parametrize("serialize_fn", serialize_fn_list)
+def test_nan_mixture(
+    distr: Distribution,
+    cat_distr: Distribution,
```

Comment:
Why do we take two distributions as input? Couldn’t these two just be two test cases? Same for p and p_cat.

Unless the test is significantly different between the categorical case vs other distributions: but in that case maybe we could split this test function into two test functions